activerecord/lib/active_record/connection_adapters/postgresql/oid/array.rb

```diff
@@ -33,7 +33,13 @@ def deserialize(value)
 
           def cast(value)
             if value.is_a?(::String)
-              value = @pg_decoder.decode(value)
+              value = begin
+                @pg_decoder.decode(value)
+              rescue TypeError
```

Comment:
I made it [a configuration option](https://github.com/ged/ruby-pg/commit/ee1e19cca83ac0b5d01b7776997964c300445e24) whether a TypeError is raised or whether a partial array is returned. The latter one is the default, so that the second commit of this PR is not strictly necessary for pg-1.1.0.